src/connection.rs

```diff
@@ -184,6 +274,7 @@ pub(crate) async fn blob_upload(
     // Send request
     let res = client
         .post(&url)
+        .query(params.as_slice())
```

Comment:
Why not just this? 🙂

```suggestion
        .query(&params)
```

Reply:
no real reason :)